ccl/changefeedccl: TestChangefeedPredicateWithSchemaChange failed

ccl/changefeedccl.TestChangefeedPredicateWithSchemaChange [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8285138?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8285138?buildTab=artifacts#/) on master @ [30d1e1c0701a385882127cf50024d3039372ff23](https://github.com/cockroachdb/cockroach/commits/30d1e1c0701a385882127cf50024d3039372ff23):


```
=== RUN   TestChangefeedPredicateWithSchemaChange
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/1b0af3c189d7f60ce633c41e08dd5486/logTestChangefeedPredicateWithSchemaChange2703349570
    test_log_scope.go:79: use -show-logs to present logs inline
=== RUN   TestChangefeedPredicateWithSchemaChange/drop_column/webhook
    helpers_test.go:751: making server as secondary tenant
    helpers_test.go:837: making webhook feed factory
=== RUN   TestChangefeedPredicateWithSchemaChange/drop_referenced_column_filter
=== RUN   TestChangefeedPredicateWithSchemaChange/drop_referenced_column_projection
=== RUN   TestChangefeedPredicateWithSchemaChange/add_alt.status
=== RUN   TestChangefeedPredicateWithSchemaChange/add_column_no_default/kafka
    helpers_test.go:751: making server as secondary tenant
    helpers_test.go:837: making kafka feed factory
=== RUN   TestChangefeedPredicateWithSchemaChange/drop_column
=== RUN   TestChangefeedPredicateWithSchemaChange/add_column/cloudstorage
    helpers_test.go:754: making server as system tenant
    helpers_test.go:837: making cloudstorage feed factory
=== RUN   TestChangefeedPredicateWithSchemaChange/add_column_no_default
=== RUN   TestChangefeedPredicateWithSchemaChange/drop_referenced_column_filter/kafka
    helpers_test.go:751: making server as secondary tenant
    helpers_test.go:837: making kafka feed factory
panic: close of closed channel

goroutine 74508 [running]:
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdceval.(*familyEvaluator).closeErr(0x0?)
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdceval/expr_eval.go:455 +0x9c
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdceval.(*Evaluator).Close(...)
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdceval/expr_eval.go:118
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*kvEventToRowConsumer).Close(0x4001de7960?)
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/event_processing.go:442 +0xc8
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*parallelEventConsumer).workerLoop.func1()
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/event_processing.go:590 +0x38
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*parallelEventConsumer).workerLoop(0x40075572c0, {0x6558410, 0x400bbd8480}, {0x65370e0, 0x400f024540}, 0x3)
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/event_processing.go:608 +0x1bc
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*parallelEventConsumer).startWorkers.func1({0x6558410?, 0x400bbd8480?})
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/event_processing.go:579 +0x40
github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1()
	github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:168 +0x2c
golang.org/x/sync/errgroup.(*Group).Go.func1()
	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:75 +0x5c
created by golang.org/x/sync/errgroup.(*Group).Go
	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:72 +0xa4
=== RUN   TestChangefeedPredicateWithSchemaChange/drop_referenced_column_projection/cloudstorage
    helpers_test.go:754: making server as system tenant
    helpers_test.go:837: making cloudstorage feed factory
=== RUN   TestChangefeedPredicateWithSchemaChange/add_alt.status/cloudstorage
    helpers_test.go:751: making server as secondary tenant
    helpers_test.go:837: making cloudstorage feed factory
=== RUN   TestChangefeedPredicateWithSchemaChange/add_column
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedPredicateWithSchemaChange.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-23365

ccl/changefeedccl.TestChangefeedPredicateWithSchemaChange [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8285611?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8285611?buildTab=artifacts#/) on master @ [73485e56768d61b14a5aa9b3f880613d7820edb9](https://github.com/cockroachdb/cockroach/commits/73485e56768d61b14a5aa9b3f880613d7820edb9):


```
=== RUN   TestChangefeedPredicateWithSchemaChange
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/1b0af3c189d7f60ce633c41e08dd5486/logTestChangefeedPredicateWithSchemaChange3357054543
    test_log_scope.go:79: use -show-logs to present logs inline
=== RUN   TestChangefeedPredicateWithSchemaChange/drop_column
=== RUN   TestChangefeedPredicateWithSchemaChange/add_alt.status/cloudstorage
    helpers_test.go:754: making server as system tenant
    helpers_test.go:837: making cloudstorage feed factory
=== RUN   TestChangefeedPredicateWithSchemaChange/add_column/kafka
    helpers_test.go:751: making server as secondary tenant
    helpers_test.go:837: making kafka feed factory
=== RUN   TestChangefeedPredicateWithSchemaChange/add_column_no_default/kafka
    helpers_test.go:754: making server as system tenant
    helpers_test.go:837: making kafka feed factory
=== RUN   TestChangefeedPredicateWithSchemaChange/drop_column/cloudstorage
    helpers_test.go:754: making server as system tenant
    helpers_test.go:837: making cloudstorage feed factory
=== RUN   TestChangefeedPredicateWithSchemaChange/drop_referenced_column_projection
=== RUN   TestChangefeedPredicateWithSchemaChange/drop_referenced_column_projection/kafka
    helpers_test.go:754: making server as system tenant
    helpers_test.go:837: making kafka feed factory
panic: close of closed channel

goroutine 83069 [running]:
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdceval.(*familyEvaluator).closeErr(0x40054f4600?)
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdceval/expr_eval.go:455 +0x9c
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdceval.(*Evaluator).Close(...)
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdceval/expr_eval.go:118
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*kvEventToRowConsumer).Close(0x40067488c0?)
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/event_processing.go:442 +0xc8
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*parallelEventConsumer).workerLoop.func1()
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/event_processing.go:590 +0x38
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*parallelEventConsumer).workerLoop(0x4005797440, {0x655a9d0, 0x400c442140}, {0x65396a0, 0x40067488c0}, 0x5)
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/event_processing.go:608 +0x1bc
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*parallelEventConsumer).startWorkers.func1({0x655a9d0?, 0x400c442140?})
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/event_processing.go:579 +0x40
github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1()
	github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:168 +0x2c
golang.org/x/sync/errgroup.(*Group).Go.func1()
	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:75 +0x5c
created by golang.org/x/sync/errgroup.(*Group).Go
	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:72 +0xa4
=== RUN   TestChangefeedPredicateWithSchemaChange/add_alt.status
=== RUN   TestChangefeedPredicateWithSchemaChange/add_column
=== RUN   TestChangefeedPredicateWithSchemaChange/add_column_no_default
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedPredicateWithSchemaChange.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/changefeedccl.TestChangefeedPredicateWithSchemaChange [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8294084?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8294084?buildTab=artifacts#/) on master @ [5bf7569902dc18d828c68971a84ebd66b9b08a2f](https://github.com/cockroachdb/cockroach/commits/5bf7569902dc18d828c68971a84ebd66b9b08a2f):


```
=== RUN   TestChangefeedPredicateWithSchemaChange
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/1b0af3c189d7f60ce633c41e08dd5486/logTestChangefeedPredicateWithSchemaChange2605919960
    test_log_scope.go:79: use -show-logs to present logs inline
=== RUN   TestChangefeedPredicateWithSchemaChange/add_column_no_default
=== RUN   TestChangefeedPredicateWithSchemaChange/drop_column
=== RUN   TestChangefeedPredicateWithSchemaChange/drop_column/cloudstorage
    helpers_test.go:754: making server as system tenant
    helpers_test.go:837: making cloudstorage feed factory
=== RUN   TestChangefeedPredicateWithSchemaChange/add_alt.status
=== RUN   TestChangefeedPredicateWithSchemaChange/add_alt.status/pubsub
    helpers_test.go:754: making server as system tenant
    helpers_test.go:837: making pubsub feed factory
=== RUN   TestChangefeedPredicateWithSchemaChange/add_column
=== RUN   TestChangefeedPredicateWithSchemaChange/add_column/enterprise
    helpers_test.go:751: making server as secondary tenant
    helpers_test.go:837: making enterprise feed factory
    helpers_test.go:851: pgURL enterprise root
=== RUN   TestChangefeedPredicateWithSchemaChange/add_column_no_default/kafka
    helpers_test.go:751: making server as secondary tenant
    helpers_test.go:837: making kafka feed factory
=== RUN   TestChangefeedPredicateWithSchemaChange/drop_referenced_column_projection
=== RUN   TestChangefeedPredicateWithSchemaChange/drop_referenced_column_projection/kafka
    helpers_test.go:751: making server as secondary tenant
    helpers_test.go:837: making kafka feed factory
panic: close of closed channel

goroutine 80845 [running]:
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdceval.(*familyEvaluator).closeErr(0x0?)
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdceval/expr_eval.go:455 +0x9c
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdceval.(*Evaluator).Close(...)
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdceval/expr_eval.go:118
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*kvEventToRowConsumer).Close(0x4002f8b7a0?)
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/event_processing.go:442 +0xc8
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*parallelEventConsumer).workerLoop.func1()
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/event_processing.go:590 +0x38
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*parallelEventConsumer).workerLoop(0x400c6955c0, {0x655c7d0, 0x400b723dc0}, {0x653b4a0, 0x4007cac000}, 0x7)
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/event_processing.go:608 +0x1bc
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*parallelEventConsumer).startWorkers.func1({0x655c7d0?, 0x400b723dc0?})
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/event_processing.go:579 +0x40
github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1()
	github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:168 +0x2c
golang.org/x/sync/errgroup.(*Group).Go.func1()
	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:75 +0x5c
created by golang.org/x/sync/errgroup.(*Group).Go
	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:72 +0xa4
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedPredicateWithSchemaChange.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/changefeedccl.TestChangefeedPredicateWithSchemaChange [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8297661?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8297661?buildTab=artifacts#/) on master @ [c3e80e0d1d34174e57990d692fa26819458508f1](https://github.com/cockroachdb/cockroach/commits/c3e80e0d1d34174e57990d692fa26819458508f1):


```
=== RUN   TestChangefeedPredicateWithSchemaChange
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/1b0af3c189d7f60ce633c41e08dd5486/logTestChangefeedPredicateWithSchemaChange2679102674
    test_log_scope.go:79: use -show-logs to present logs inline
=== RUN   TestChangefeedPredicateWithSchemaChange/add_alt.status
=== RUN   TestChangefeedPredicateWithSchemaChange/add_alt.status/kafka
    helpers_test.go:775: making server as secondary tenant
    helpers_test.go:861: making kafka feed factory
=== RUN   TestChangefeedPredicateWithSchemaChange/add_column/webhook
    helpers_test.go:775: making server as secondary tenant
    helpers_test.go:861: making webhook feed factory
=== RUN   TestChangefeedPredicateWithSchemaChange/add_column_no_default/kafka
    helpers_test.go:775: making server as secondary tenant
    helpers_test.go:861: making kafka feed factory
=== RUN   TestChangefeedPredicateWithSchemaChange/drop_column/pubsub
    helpers_test.go:775: making server as secondary tenant
    helpers_test.go:861: making pubsub feed factory
=== RUN   TestChangefeedPredicateWithSchemaChange/drop_referenced_column_filter
=== RUN   TestChangefeedPredicateWithSchemaChange/drop_referenced_column_projection/webhook
    helpers_test.go:778: making server as system tenant
    helpers_test.go:861: making webhook feed factory
=== RUN   TestChangefeedPredicateWithSchemaChange/add_column
=== RUN   TestChangefeedPredicateWithSchemaChange/add_column_no_default
=== RUN   TestChangefeedPredicateWithSchemaChange/drop_column
=== RUN   TestChangefeedPredicateWithSchemaChange/drop_referenced_column_filter/kafka
    helpers_test.go:775: making server as secondary tenant
    helpers_test.go:861: making kafka feed factory
panic: close of closed channel

goroutine 85271 [running]:
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdceval.(*familyEvaluator).closeErr(0x0?)
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdceval/expr_eval.go:465 +0x9c
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdceval.(*Evaluator).Close(...)
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdceval/expr_eval.go:120
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*kvEventToRowConsumer).Close(0x40021d61c0?)
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/event_processing.go:468 +0xc8
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*parallelEventConsumer).workerLoop.func1()
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/event_processing.go:616 +0x38
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*parallelEventConsumer).workerLoop(0x4008cf35c0, {0x65710f0, 0x40068d1b00}, {0x654fd20, 0x4009cbcd20}, 0x7)
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/event_processing.go:634 +0x1bc
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*parallelEventConsumer).startWorkers.func1({0x65710f0?, 0x40068d1b00?})
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/event_processing.go:605 +0x40
github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1()
	github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:168 +0x2c
golang.org/x/sync/errgroup.(*Group).Go.func1()
	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:75 +0x5c
created by golang.org/x/sync/errgroup.(*Group).Go
	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:72 +0xa4
=== RUN   TestChangefeedPredicateWithSchemaChange/drop_referenced_column_projection
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedPredicateWithSchemaChange.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/changefeedccl.TestChangefeedPredicateWithSchemaChange [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_UnitTests_BazelUnitTests/8305274?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_UnitTests_BazelUnitTests/8305274?buildTab=artifacts#/) on master @ [a3d53668ff52c2306f37fcad8e9212ff665ca115](https://github.com/cockroachdb/cockroach/commits/a3d53668ff52c2306f37fcad8e9212ff665ca115):


```
=== RUN   TestChangefeedPredicateWithSchemaChange
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/3fdc1e0236634cdb58a612313ff148df/logTestChangefeedPredicateWithSchemaChange4279414571
    test_log_scope.go:79: use -show-logs to present logs inline
=== RUN   TestChangefeedPredicateWithSchemaChange/drop_column/enterprise
    helpers_test.go:775: making server as secondary tenant
    helpers_test.go:861: making enterprise feed factory
    helpers_test.go:875: pgURL enterprise root
=== RUN   TestChangefeedPredicateWithSchemaChange/drop_referenced_column_filter
=== RUN   TestChangefeedPredicateWithSchemaChange/drop_referenced_column_filter/pubsub
    helpers_test.go:778: making server as system tenant
    helpers_test.go:861: making pubsub feed factory
panic: close of closed channel

goroutine 76634 [running]:
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdceval.(*familyEvaluator).closeErr(0xc004dc4740?)
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdceval/expr_eval.go:465 +0xa3
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdceval.(*Evaluator).Close(...)
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdceval/expr_eval.go:120
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*kvEventToRowConsumer).Close(0xc0082b4780?)
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/event_processing.go:468 +0xe8
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*parallelEventConsumer).workerLoop.func1()
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/event_processing.go:616 +0x38
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*parallelEventConsumer).workerLoop(0xc00ca25c80, {0x6d7abd0, 0xc0084bfa40}, {0x6d59910, 0xc0082b4780}, 0x5)
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/event_processing.go:634 +0x291
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*parallelEventConsumer).startWorkers.func1({0x6d7abd0?, 0xc0084bfa40?})
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/event_processing.go:605 +0x3c
github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1()
	github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:168 +0x25
golang.org/x/sync/errgroup.(*Group).Go.func1()
	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:75 +0x64
created by golang.org/x/sync/errgroup.(*Group).Go
	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:72 +0xa5
=== RUN   TestChangefeedPredicateWithSchemaChange/drop_referenced_column_projection
=== RUN   TestChangefeedPredicateWithSchemaChange/add_alt.status
=== RUN   TestChangefeedPredicateWithSchemaChange/add_column_no_default/pubsub
    helpers_test.go:775: making server as secondary tenant
    helpers_test.go:861: making pubsub feed factory
=== RUN   TestChangefeedPredicateWithSchemaChange/drop_column
=== RUN   TestChangefeedPredicateWithSchemaChange/drop_referenced_column_projection/cloudstorage
    helpers_test.go:775: making server as secondary tenant
    helpers_test.go:861: making cloudstorage feed factory
=== RUN   TestChangefeedPredicateWithSchemaChange/add_alt.status/cloudstorage
    helpers_test.go:778: making server as system tenant
    helpers_test.go:861: making cloudstorage feed factory
=== RUN   TestChangefeedPredicateWithSchemaChange/add_column
=== RUN   TestChangefeedPredicateWithSchemaChange/add_column/webhook
    helpers_test.go:775: making server as secondary tenant
    helpers_test.go:861: making webhook feed factory
=== RUN   TestChangefeedPredicateWithSchemaChange/add_column_no_default
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedPredicateWithSchemaChange.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/changefeedccl.TestChangefeedPredicateWithSchemaChange [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_UnitTests_BazelUnitTests/8305274?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_UnitTests_BazelUnitTests/8305274?buildTab=artifacts#/) on master @ [a3d53668ff52c2306f37fcad8e9212ff665ca115](https://github.com/cockroachdb/cockroach/commits/a3d53668ff52c2306f37fcad8e9212ff665ca115):


```
=== RUN   TestChangefeedPredicateWithSchemaChange
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/3fdc1e0236634cdb58a612313ff148df/logTestChangefeedPredicateWithSchemaChange271867263
    test_log_scope.go:79: use -show-logs to present logs inline
=== RUN   TestChangefeedPredicateWithSchemaChange/add_column/cloudstorage
    helpers_test.go:775: making server as secondary tenant
    helpers_test.go:861: making cloudstorage feed factory
=== RUN   TestChangefeedPredicateWithSchemaChange/add_column_no_default/kafka
    helpers_test.go:778: making server as system tenant
    helpers_test.go:861: making kafka feed factory
=== RUN   TestChangefeedPredicateWithSchemaChange/drop_referenced_column_filter/cloudstorage
    helpers_test.go:775: making server as secondary tenant
    helpers_test.go:861: making cloudstorage feed factory
panic: close of closed channel

goroutine 109732 [running]:
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdceval.(*familyEvaluator).closeErr(0x0?)
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdceval/expr_eval.go:465 +0xa3
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdceval.(*Evaluator).Close(...)
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdceval/expr_eval.go:120
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*kvEventToRowConsumer).Close(0xc0053721c0?)
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/event_processing.go:468 +0xe8
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*parallelEventConsumer).workerLoop.func1()
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/event_processing.go:616 +0x38
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*parallelEventConsumer).workerLoop(0xc0080acfc0, {0x6d7abd0, 0xc009c959c0}, {0x6d59910, 0xc00940e5a0}, 0x7)
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/event_processing.go:634 +0x291
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*parallelEventConsumer).startWorkers.func1({0x6d7abd0?, 0xc009c959c0?})
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/event_processing.go:605 +0x3c
github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1()
	github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:168 +0x25
golang.org/x/sync/errgroup.(*Group).Go.func1()
	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:75 +0x64
created by golang.org/x/sync/errgroup.(*Group).Go
	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:72 +0xa5
=== RUN   TestChangefeedPredicateWithSchemaChange/drop_referenced_column_projection
=== RUN   TestChangefeedPredicateWithSchemaChange/add_alt.status
=== RUN   TestChangefeedPredicateWithSchemaChange/add_column
=== RUN   TestChangefeedPredicateWithSchemaChange/drop_column/kafka
    helpers_test.go:775: making server as secondary tenant
    helpers_test.go:861: making kafka feed factory
=== RUN   TestChangefeedPredicateWithSchemaChange/drop_referenced_column_filter
=== RUN   TestChangefeedPredicateWithSchemaChange/drop_referenced_column_projection/cloudstorage
    helpers_test.go:778: making server as system tenant
    helpers_test.go:861: making cloudstorage feed factory
=== RUN   TestChangefeedPredicateWithSchemaChange/add_alt.status/pubsub
    helpers_test.go:775: making server as secondary tenant
    helpers_test.go:861: making pubsub feed factory
=== RUN   TestChangefeedPredicateWithSchemaChange/add_column_no_default
=== RUN   TestChangefeedPredicateWithSchemaChange/drop_column
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedPredicateWithSchemaChange.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/changefeedccl.TestChangefeedPredicateWithSchemaChange [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8305701?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8305701?buildTab=artifacts#/) on master @ [59e7cabb81ed06f7011dee8a76abc7a7edb5dfc7](https://github.com/cockroachdb/cockroach/commits/59e7cabb81ed06f7011dee8a76abc7a7edb5dfc7):


```
=== RUN   TestChangefeedPredicateWithSchemaChange
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/1b0af3c189d7f60ce633c41e08dd5486/logTestChangefeedPredicateWithSchemaChange4204183572
    test_log_scope.go:79: use -show-logs to present logs inline
=== RUN   TestChangefeedPredicateWithSchemaChange/add_column
=== RUN   TestChangefeedPredicateWithSchemaChange/add_column/cloudstorage
    helpers_test.go:775: making server as secondary tenant
    helpers_test.go:861: making cloudstorage feed factory
=== RUN   TestChangefeedPredicateWithSchemaChange/drop_column/cloudstorage
    helpers_test.go:775: making server as secondary tenant
    helpers_test.go:861: making cloudstorage feed factory
=== RUN   TestChangefeedPredicateWithSchemaChange/drop_referenced_column_projection
=== RUN   TestChangefeedPredicateWithSchemaChange/drop_column
=== RUN   TestChangefeedPredicateWithSchemaChange/drop_referenced_column_projection/enterprise
    helpers_test.go:778: making server as system tenant
    helpers_test.go:861: making enterprise feed factory
    helpers_test.go:875: pgURL enterprise root
panic: close of closed channel

goroutine 74239 [running]:
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdceval.(*familyEvaluator).closeErr(0x400b8302c0?)
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdceval/expr_eval.go:465 +0x9c
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdceval.(*Evaluator).Close(...)
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdceval/expr_eval.go:120
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*kvEventToRowConsumer).Close(0x400a009c20?)
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/event_processing.go:468 +0xc8
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*parallelEventConsumer).workerLoop.func1()
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/event_processing.go:616 +0x38
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*parallelEventConsumer).workerLoop(0x4006554a80, {0x6578ab0, 0x40088a6180}, {0x6557680, 0x400a009c20}, 0x1)
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/event_processing.go:634 +0x1bc
github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*parallelEventConsumer).startWorkers.func1({0x6578ab0?, 0x40088a6180?})
	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/event_processing.go:605 +0x40
github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1()
	github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:168 +0x2c
golang.org/x/sync/errgroup.(*Group).Go.func1()
	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:75 +0x5c
created by golang.org/x/sync/errgroup.(*Group).Go
	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:72 +0xa4
=== RUN   TestChangefeedPredicateWithSchemaChange/add_alt.status
=== RUN   TestChangefeedPredicateWithSchemaChange/add_alt.status/kafka
    helpers_test.go:775: making server as secondary tenant
    helpers_test.go:861: making kafka feed factory
=== RUN   TestChangefeedPredicateWithSchemaChange/add_column_no_default
=== RUN   TestChangefeedPredicateWithSchemaChange/add_column_no_default/cloudstorage
    helpers_test.go:775: making server as secondary tenant
    helpers_test.go:861: making cloudstorage feed factory
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedPredicateWithSchemaChange.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/ccl/changefeedccl/cdceval/expr_eval.go
+++ b/pkg/ccl/changefeedccl/cdceval/expr_eval.go
@@ -450,7 +450,11 @@ func (e *familyEvaluator) setupContextForRow(ctx context.Context, updated cdceve
 
 func (e *familyEvaluator) closeErr() error {
 	if e.errCh != nil {
-		defer close(e.errCh) // Must be deferred since planGroup  go routine might write.
+		// Must be deferred since planGroup  go routine might write.
+		defer func() {
+			close(e.errCh)
+			e.errCh = nil
+		}()
 	}
 
 	if e.input != nil {

